live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/domain/utils/CollectionUtils.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * #%L
+ * Alfresco HX Insight Connector
+ * %%
+ * Copyright (C) 2023 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+
+package org.alfresco.hxi_connector.live_ingester.domain.utils;
+
+import static java.util.function.Predicate.not;
+
+import static lombok.AccessLevel.PRIVATE;
+
+import java.util.Collection;
+import java.util.List;
+
+import lombok.NoArgsConstructor;
+
+@NoArgsConstructor(access = PRIVATE)
+public final class CollectionUtils
+{
+    public static <T> List<T> difference(Collection<T> first, Collection<T> second)
```

Comment:
It would be worth adding some javadoc to clarify that this is `first - second` since it's possible to have the arguments misordered and it's also fairly common to use "difference" when meaning symmetric difference.